src/smc-webapp/course/handouts_panel.cjsx

```diff
@@ -0,0 +1,563 @@
+                items         = {@props.all_handouts}
+                add_folders   = {(paths)=>paths.map(@props.actions.add_handout)}
+                item_name     = {"handout"}
+                plural_item_name = {"handouts"}
+            />
+
+        <Panel header={header}>
+            {for handout, i in shown_handouts
+                <Handout backgroundColor={if i%2==0 then "#eee"}  key={handout.handout_id}
+                        handout={@props.all_handouts.get(handout.handout_id)} project_id={@props.project_id}
+                        students={@props.students} user_map={@props.user_map} actions={@props.actions}
+                        store={@props.store} open_directory={@props.project_actions.open_directory}
+                />}
+            {@render_show_deleted_button(num_deleted) if num_deleted > 0}
+        </Panel>
+
+exports.HandoutsPanel.Header = rclass
+    propTypes :
+        n : rtypes.number
+
+    render: ->
+        <Tip delayShow=1300
+             title="Handouts"
+             tip="This tab lists all of the handouts associated with your course.">
+            <span>
+                <Icon name="files-o"/> Handouts {if @props.n? then " (#{@props.n})" else ""}
+            </span>
+        </Tip>
+
+Handout = rclass
+    propTypes :
+        handout             : rtypes.object
+        backgroundColor     : rtypes.string
+        store               : rtypes.object
+        actions             : rtypes.object
+        open_directory      : rtypes.func     # open_directory(path)
+
+    getInitialState : ->
+        more : false
+        confirm_delete : false
+
+    open_handout_path : (e)->
+        e.preventDefault()
+        @props.open_directory(@props.handout.get('path'))
+
+    copy_handout_to_all : (step, new_only) ->
+        @props.actions.copy_handout_to_all_students(@props.handout, new_only)
+
+    render_more_header : ->
+        <div>
+            <Button onClick={@open_handout_path}>
+                <Icon name="folder-open-o" /> Edit Handout
+            </Button>
+        </div>
+
+    render_handout_notes : ->
+        <Row key='note' style={styles.note}>
+            <Col xs=2>
+                <Tip title="Notes about this handout" tip="Record notes about this handout here. These notes are only visible to you, not to your students.  Put any instructions to students about handouts in a file in the directory that contains the handout.">
+                    Private Handout Notes<br /><span style={color:"#666"}></span>
+                </Tip>
+            </Col>
+            <Col xs=10>
+                <MarkdownInput
+                    rows          = 6
+                    placeholder   = 'Private notes about this handout (not visible to students)'
+                    default_value = {@props.handout.get('note')}
+                    on_save       = {(value)=>@props.actions.set_handout_note(@props.handout, value)}
+                />
+            </Col>
+        </Row>
+
```

Comment:
The small render methods after this comment could be moved off since it mostly shares with the same code in assignments. However, it is slightly different enough that I prefer to keep it as is until we have at least a third area where we have the same code use.